rayon-core/src/configuration.rs

```diff
@@ -77,7 +58,7 @@ impl Configuration {
 
     /// Get the number of threads that will be used for the thread
     /// pool. See `set_num_threads` for more information.
-    pub fn num_threads(&self) -> Option<usize> {
+    pub fn num_threads(&self) -> usize {
         self.num_threads
```

Comment:
@cuviper What about setting `num_threads` eagerly by moving the logic to `new()` and then have `set_num_threads` override it (if argument not zero)?